### PR TITLE
BUG: secondary y axis could not be set to log scale (#25545)

### LINKED ISSUE #25545: Plotting: secondary y axis can not be set to log scale
#### Code Sample, a copy-pastable example if possible

```python
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt

pd.Series(np.random.random(1000)).plot(logy=True)
pd.Series(np.random.random(1000)).plot(secondary_y=True, logy=True)
plt.show()
```

#### Problem description

When I plot some data and set both the `secondary_y` and `logy` parameters to `True`, the `logy` parameter is not respected. Instead, the secondary y axis still uses linear scaling.

logtest

#### Expected Output

The secondary y axis should use logarithmic scaling.

logtest

#### Output of ``pd.show_versions()``

<details>

INSTALLED VERSIONS
------------------
commit: None
python: 3.7.2.final.0
python-bits: 64
OS: Linux
OS-release: 4.19.0-2-amd64
machine: x86_64
processor: 
byteorder: little
LC_ALL: None
LANG: de_DE.UTF-8
LOCALE: de_DE.UTF-8
pandas: 0.24.1
pytest: 4.1.1
pip: 18.1
setuptools: 40.6.3
Cython: None
numpy: 1.16.1
scipy: 1.1.0
pyarrow: None
xarray: None
IPython: 5.8.0
sphinx: 1.8.3
patsy: None
dateutil: 2.7.3
pytz: 2018.9
blosc: None
bottleneck: None
tables: 3.4.4
numexpr: 2.6.9
feather: None
matplotlib: 3.0.2
openpyxl: None
xlrd: None
xlwt: None
xlsxwriter: None
lxml.etree: 4.3.1
bs4: 4.7.1
html5lib: 1.0.1
sqlalchemy: None
pymysql: None
psycopg2: None
jinja2: 2.10
s3fs: None
fastparquet: None
pandas_gbq: None
pandas_datareader: None
gcsfs: None

</details>

### PR DESCRIPTION
**Pull request by johan12345**
- [x] closes #25545
- [x] tests added / passed
- [x] passes `git diff upstream/master -u -- "*.py" | flake8 --diff`
- [x] whatsnew entry

### CODE REVIEW

**Comment by johan12345** [comment]
Codecov seems to think that my new line of code
```
new_ax.set_yscale('log')
```
was not covered by the tests. Is this because tests marked with `@pytest.mark.slow` are not included in the coverage calculations?

File: pandas/plotting/_core.py [hunk 1]
```diff
@@ -287,6 +287,9 @@ def _maybe_right_yaxis(self, ax, axes_num):
 
             if not self._has_plotted_object(orig_ax):  # no data on left y
                 orig_ax.get_yaxis().set_visible(False)
+
+            if self.logy or self.loglog:
```
Comment (TomAugspurger) [inline pandas/plotting/_core.py]: Just FYI @charlesdong1991 we'll want to update https://github.com/pandas-dev/pandas/pull/24968 to handle this new case.

This PR is being back ported to 0.24.2. After it's in master, we'll want to think about consolidating all this "axis log" stuff.

**Review by TomAugspurger — APPROVED** [review]

**Comment by TomAugspurger** [comment]
You can probably ignore codecov on this one. It tricky to know exactly which tests are run with which dependencies on our coverage runs.

**Comment by gfyoung** [comment]
@TomAugspurger : This PR is green.  Are we still porting to `0.24.2` ?  (I noticed you didn't label as such, so wasn't sure)

**Comment by TomAugspurger** [comment]
Yep, forgot to set the milestone.

Thanks @johan12345!

**Comment by johan12345** [comment]
Thank you!

**Pull request merged**